src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
+const BASE_HEALTH = 100;
+const BASE_ATTACK = 50;
+const MIN_HEALTH = 0;
+
 class Animal {
-  // write your code here
+  constructor(name, health = BASE_HEALTH) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
```

Comment:
There is no need to do it here
Animal will be pushed when you do `super` call

The same below